docs/specification.md

```diff
@@ -27,6 +27,69 @@ These two parts are coupled into one smart contract because interaction between
 - The storage of the contract must have annotations for all fields
   and must be documented to make its interpretation easy for users.
 
+# Configuration
+
+BaseDAO allows configuration via:
+
+```haskell
+type Parameter = BaseDAO.Parameter Config ProposalMetadata
+```
+
+The `Config` type should have this shape:
+
+```haskell
+data Config = Config
+  { cDaoName :: Text
+  , cDaoDescription :: Text
+  , cProposalBondCalculation :: Lambda
+    -- ^ can be constant value: `frozen_token`
+    -- can be a lambda that varies depend on a property of proposal.
+  , cVotingPeriod :: Natural
+    -- ^ TODO: which type is preferred?
+  , cQuorumThreshold :: Natural
+  , cContractMetadata :: ContractMetadata
+  , cDecisionLambda :: _Hole
+  , cProposalRejectionLogic :: _Hole
+      -- ^ TODO: Quote: "Proposals can be rejected by arbitrary logic
+      -- Example: ensuring a minimum or maximum of a treasury
+      -- amount specified in a funding-based DAO"
+  , cTrustedAdmin :: Address
+  ...
+  }
+
+-- From TZIP-16: https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md
+data ContractMetadata = ContractMetadata
+  { cmFrozenTokenMetadata :: _Hole
+      -- ^ TODO: Quote: "The FA2 metadata for the frozen token
+      -- should include the DAO metadata."
+  -- TODO: add more fields
+  }
+```
+
+`ProposalMetadata` contains fields that are needed to propose a proposal.
+This can contain any fields. Here is an example of "treasury" style DAO.
+```haskell
+data ProposalMetadata = ProposalMetadata
+  { mutezAmount :: Mutez
+  , recipientId :: Text
+  , agoraPostId :: Natural
+  ...
+  }
+```
+
+TODO: What is Associated FA2 Contract:
```

Comment:
The associated FA2 contract is SELF. As stated above, FA2 logic and DAO logic are coupled into one smart contract.